src/components/application_manager/test/resumption/resume_ctrl_test.cc

```diff
@@ -705,6 +728,9 @@ TEST_F(ResumeCtrlTest, SetAppHMIState_HMIFull_WithPolicy_DevDisallowed) {
 TEST_F(ResumeCtrlTest, SaveAllApplications) {
   utils::SharedPtr<application_manager::Application> app_sh_mock =
       ::utils::MakeShared<application_manager_test::MockApplication>();
+  EXPECT_CALL(*static_cast<application_manager_test::MockApplication*>(
```

Comment:
@vlantonov I insist on creation SharedPtr to MockApp and the use testing::Eq(mock_app). In this case you don't need any cast.